include/gridtools/stencil_composition/icosahedral_grids/backend_x86/mss_loop_x86.hpp

```diff
@@ -31,6 +33,28 @@ namespace gridtools {
                 (stage_group_contains_color<Color>::template apply, GT_META_CALL(meta::at_c, (T, 2))));
         };
 
+        template <uint_t Color>
+        struct run_esf_functor_x86 {
+            template <class StageGroups, class ItDomain>
+            GT_FORCE_INLINE static void exec(ItDomain &it_domain) {
+                using stages_t = GT_META_CALL(meta::flatten, StageGroups);
+                GT_STATIC_ASSERT(meta::length<stages_t>::value == 1, GT_INTERNAL_ERROR);
+                using stage_t = GT_META_CALL(meta::first, stages_t);
+                stage_t::template exec<Color>(it_domain);
+            }
+        };
+
+        template <class LoopIntervals>
+        struct get_ncolors;
+
+        template <template <class...> class L0,
+            template <class...> class L1,
+            template <class...> class L2,
+            class From,
+            class To,
+            class Stage>
+        struct get_ncolors<L0<loop_interval<From, To, L1<L2<Stage>>>>> : Stage::n_colors {};
```

Comment:
This looks a bit weird. Is there no better way to get the stage?

Reply:
I add a comment their